signalfx_lambda/metrics.py

```diff
@@ -0,0 +1,132 @@
+import functools
```

Comment:
This file was pretty much extracted wholesale from `__init__.py`, no significant changes were made